### PR TITLE
Document Artifacts

### LINKED ISSUE #48: Store build artifacts
I recently tried Cirrus-CI after coming from Circle-CI.
One feature I'm missing (or at least I have not found it) is storing artifacts (for reference: [Circle-CI docs](https://circleci.com/docs/2.0/artifacts/)) after a build.

In my case I would like to store the `build/reports` folder of a `Gradle` build and be able to access it after the build through the web frontend. This would be especially usefull if my tests or static code analysis tools failed and the error is not printed to the console, since I have no way to see them without running the build locally.

Thanks in advance!

### PR DESCRIPTION
**Pull request by fkorotkov**
Fixes #48

### CODE REVIEW

File: docs/guide/writing-tasks.md [hunk 1]
```diff
@@ -152,6 +152,27 @@ test_task:
...
+An `artifacts` instruction allows to store files and expose them in the UI for downloading later. An `artifacts` instruction 
+can be named the same way as `script` instruction and has only one required `path` field which accepts a [glob pattern](https://en.wikipedia.org/wiki/Glob_(programming))
+of files to store.
+
+In the example below, *Build and Test* task produces two artifacts: `binaries` artifacts with all executables built during a 
+successful task completion and `junit` artifacts with all test reports regardless of the final task status (more about 
+that you can learn in the [next section describing execution behaviour](#execution-behavior-of-instructions)).
+
+```yaml
+task:
```
Comment (RDIL) [inline docs/guide/writing-tasks.md]:
```suggestion
build_and_test_task:
```

File: docs/guide/writing-tasks.md [hunk 2]
```diff
@@ -152,6 +152,27 @@ test_task:
...
+  name: Build and Test
```
Comment (RDIL) [inline docs/guide/writing-tasks.md]:
```suggestion
```

**Comment by RDIL** [comment]
@fkorotkov this seems to break it:
```yaml
unittest_results_artifacts:
  path: *.xml
```

**Comment by RDIL** [comment]
@fkorotkov also the download link doesn't work:
https://api.cirrus-ci.com/v1/artifact/task/5954095255912448/unittest_results//TEST-tests.TestCode-20190326232611.xml

**Comment by fkorotkov** [comment]
@RDIL fixed in https://github.com/cirruslabs/cirrus-ci-web/commit/72db5b580b2f6f1ff2e5b6fceecf7f02da02afce. Thanks for testing! 🙌

**Pull request merged**